### PR TITLE
MAINT: Get rid of NumarrayTypes.

### PR DESCRIPTION
**Pull request by jaimefrio**

### CODE REVIEW

**Comment by rgommers** [comment]
The CircleCI build failed:

    creating build/temp.linux-x86_64-2.7/scipy/ndimage
    creating build/temp.linux-x86_64-2.7/scipy/ndimage/src
    compile options: '-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION -DATLAS_INFO="\"3.10.1\"" -Iscipy/ndimage/src -I/home/ubuntu/virtualenvs/venv-system/lib/python2.7/site-packages/numpy/core/include -Iscipy/_lib/src -I/home/ubuntu/virtualenvs/venv-system/lib/python2.7/site-packages/numpy/core/include -I/opt/circleci/python/2.7.11/include/python2.7 -c'
    gcc: scipy/ndimage/src/ni_filters.c
    In file included from scipy/ndimage/src/ni_filters.c:32:0:
    scipy/ndimage/src/ni_support.h:102:39: error: unknown type name ‘UInt32’
     int NI_SubspaceIterator(NI_Iterator*, UInt32);
                                           ^
    scipy/ndimage/src/ni_support.h:231:42: error: unknown type name ‘Bool’
     int NI_InitFilterOffsets(PyArrayObject*, Bool*, npy_intp*,
                                              ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_Correlate1D’:
    scipy/ndimage/src/ni_filters.c:45:5: error: unknown type name ‘Float64’
         Float64 *fw;
         ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_Correlate’:
    scipy/ndimage/src/ni_filters.c:163:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:169:5: error: unknown type name ‘Float64’
         Float64 *pw;
         ^
    scipy/ndimage/src/ni_filters.c:170:5: error: unknown type name ‘Float64’
         Float64 *ww = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:176:11: error: ‘Float64’ undeclared (first use in this function)
         pw = (Float64*)PyArray_DATA(weights);
               ^
    scipy/ndimage/src/ni_filters.c:176:11: note: each undeclared identifier is reported only once for each function it appears in
    scipy/ndimage/src/ni_filters.c:176:19: error: expected expression before ‘)’ token
         pw = (Float64*)PyArray_DATA(weights);
                       ^
    scipy/ndimage/src/ni_filters.c:177:32: error: ‘Bool’ undeclared (first use in this function)
         pf = malloc(fsize * sizeof(Bool));
                                    ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_MinOrMaxFilter’:
    scipy/ndimage/src/ni_filters.c:534:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:542:5: error: unknown type name ‘Float64’
         Float64 *ps;
         ^
    scipy/ndimage/src/ni_filters.c:547:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:547:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    scipy/ndimage/src/ni_filters.c:561:15: error: ‘Float64’ undeclared (first use in this function)
             ps = (Float64*)PyArray_DATA(structure);
                   ^
    scipy/ndimage/src/ni_filters.c:561:23: error: expected expression before ‘)’ token
             ps = (Float64*)PyArray_DATA(structure);
                           ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_RankFilter’:
    scipy/ndimage/src/ni_filters.c:733:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:740:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:740:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_GenericFilter’:
    scipy/ndimage/src/ni_filters.c:942:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:952:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:952:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    In file included from scipy/ndimage/src/ni_filters.c:32:0:
    scipy/ndimage/src/ni_support.h:102:39: error: unknown type name ‘UInt32’
     int NI_SubspaceIterator(NI_Iterator*, UInt32);
                                           ^
    scipy/ndimage/src/ni_support.h:231:42: error: unknown type name ‘Bool’
     int NI_InitFilterOffsets(PyArrayObject*, Bool*, npy_intp*,
                                              ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_Correlate1D’:
    scipy/ndimage/src/ni_filters.c:45:5: error: unknown type name ‘Float64’
         Float64 *fw;
         ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_Correlate’:
    scipy/ndimage/src/ni_filters.c:163:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:169:5: error: unknown type name ‘Float64’
         Float64 *pw;
         ^
    scipy/ndimage/src/ni_filters.c:170:5: error: unknown type name ‘Float64’
         Float64 *ww = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:176:11: error: ‘Float64’ undeclared (first use in this function)
         pw = (Float64*)PyArray_DATA(weights);
               ^
    scipy/ndimage/src/ni_filters.c:176:11: note: each undeclared identifier is reported only once for each function it appears in
    scipy/ndimage/src/ni_filters.c:176:19: error: expected expression before ‘)’ token
         pw = (Float64*)PyArray_DATA(weights);
                       ^
    scipy/ndimage/src/ni_filters.c:177:32: error: ‘Bool’ undeclared (first use in this function)
         pf = malloc(fsize * sizeof(Bool));
                                    ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_MinOrMaxFilter’:
    scipy/ndimage/src/ni_filters.c:534:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:542:5: error: unknown type name ‘Float64’
         Float64 *ps;
         ^
    scipy/ndimage/src/ni_filters.c:547:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:547:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    scipy/ndimage/src/ni_filters.c:561:15: error: ‘Float64’ undeclared (first use in this function)
             ps = (Float64*)PyArray_DATA(structure);
                   ^
    scipy/ndimage/src/ni_filters.c:561:23: error: expected expression before ‘)’ token
             ps = (Float64*)PyArray_DATA(structure);
                           ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_RankFilter’:
    scipy/ndimage/src/ni_filters.c:733:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:740:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:740:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    scipy/ndimage/src/ni_filters.c: In function ‘NI_GenericFilter’:
    scipy/ndimage/src/ni_filters.c:942:5: error: unknown type name ‘Bool’
         Bool *pf = NULL;
         ^
    scipy/ndimage/src/ni_filters.c:952:11: error: ‘Bool’ undeclared (first use in this function)
         pf = (Bool*)PyArray_DATA(footprint);
               ^
    scipy/ndimage/src/ni_filters.c:952:16: error: expected expression before ‘)’ token
         pf = (Bool*)PyArray_DATA(footprint);
                    ^
    error: Command "gcc -pthread -fno-strict-aliasing -g -O2 -O0 -ggdb -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION -DATLAS_INFO="\"3.10.1\"" -Iscipy/ndimage/src -I/home/ubuntu/virtualenvs/venv-system/lib/python2.7/site-packages/numpy/core/include -Iscipy/_lib/src -I/home/ubuntu/virtualenvs/venv-system/lib/python2.7/site-packages/numpy/core/include -I/opt/circleci/python/2.7.11/include/python2.7 -c scipy/ndimage/src/ni_filters.c -o build/temp.linux-x86_64-2.7/scipy/ndimage/src/ni_filters.o" failed with exit status 1
    
    Build failed!

**Comment by jaimefrio** [comment]
I suppose my local tests did not rebuild everything, will clean and fix it.

**Comment by jaimefrio** [comment]
This last commit solves it for me locally...

File: scipy/ndimage/src/ni_morphology.c [hunk 1]
```diff
@@ -703,14 +703,14 @@ int NI_DistanceTransformBruteForce(PyArrayObject* input, int metric,
                     temp = temp->next;
                 }
                 if (distances)
-                    *(UInt32*)pd = distance;
+                    *(npy_int32*)pd = distance;
```
Comment (rgommers) [inline scipy/ndimage/src/ni_morphology.c]: This looks correct, but just checking: you changed ``uint32`` to ``int32`` on purpose here right?

Comment (rgommers) [inline scipy/ndimage/src/ni_morphology.c]: I see further down there's much more of the same, never mind.

**Comment by rgommers** [comment]
LGTM, merged. Thanks Jaime

**Pull request merged**